cockroachcloud/cloud-sso-sql.md

```diff
@@ -0,0 +1,147 @@
+
+	By default, your cluster's configuration will contain the {{ site.data.products.db }}'s own public key, allowing Cockroach Cloud to serve as an IdP.
+
+	When modifying this cluster setting, you must include the Cockroach Cloud public key in the key set. Failing to do so can prevent maintenance access by essential Cockroach Cloud managed service accounts, leading to unintended consequences.
+
+	!!!{ @cameron Fact check on this? Seems right}
+
+1. `server.jwt_authentication.issuers`
+
+	Add your IdP's formal `issuer` name (this must match the `issuer` field in the JWT itself) to your cluster's list of accepted token issuers:
+
+	A comma separated list of formal names of accepted JWT issuers. This list must include a given IdP, or the cluster will reject JWTs issued by it.
+
+	By default will be the CC issuer but they can set it to any string or any list of strings formatted as a json array. Needs to match the issuer of JWTs
+
+1. `server.jwt_authentication.audience`
+	
+	The name of your cluster as specified by the IdP, or a comma-separated list of such names. One of the audience names must match the `audience` field with which your IdP will generate JWT formatted auth tokens.
+
+1. `server.jwt_authentication.enabled`
+
+	Required value: `true`
+
+	Check to confirm that JWT authentication is enabled on your cluster. It is enabled by default in {{ site.data.products.db }} clusters.
+
+### Authenticate to your cluster with your JWT token
+
+To provision SQL cluster access for service accounts, you must provision OIDC or SAML tokens. There are many ways to do this, which are beyond the scope of this tutorial.
+
+For example, your Google Cloud Platform organization can serve as IdP by issuing OIDC auth tokens, as described here in the [GCP docs on issuing tokens to service accounts](https://cloud.google.com/iam/docs/create-short-lived-credentials-direct#sa-credentials-oidc). This [blog post](https://morgans-blog.deno.dev/sso-crdb-gcp) discussing using GCP-issued OIDC tokens to authenticate to CockroachDB.
+
+Once you have a valid JWT auth token (with `issuer` and `audience` matching the values [configured in your cluster settings](#configure-your-cluster-to-accept-your-external-identity-provider)) from your IdP, you may use it to connect to your cluster's SQL interface.
+
+{{site.data.alerts.success}}
+This example uses [`cockroach sql`](../{{site.versions["stable"]}}/cockroach-sql.html), but you can use any SQL client that supports sufficiently long passwords.
+{{site.data.alerts.end}}
+
+{% include_cached copy-clipboard.html %}
+~~~shell
+cockroach sql --url "postgresql://{SQL_USERNAME}:{JWT_TOKEN}@{CLUSTER_HOST}:26257?options=--crdb:jwt_auth_enabled=true" --certs-dir={CLUSTER_CERT_DIR}
+~~~
+
+~~~txt
+Welcome to the cockroach SQL interface...
+~~~
+
+## Support for tokens with SQL-incompatible subject names
```

Comment:
Did you intend for this heading to be a H2? It seems to not quite fit with the page structure.

Reply:
It's less of a full section, but it doesn't nest under the others, so yeah, it gets its own h2 as far as I can see.